pylint/checkers/spelling.py

```diff
@@ -378,7 +383,10 @@ def _check_spelling(self, msgid, line, line_num):
 
             # Store word to private dict or raise a message.
             if self.linter.config.spelling_store_unknown_words:
-                if lower_cased_word not in self.unknown_words:
+                if (
+                    lower_cased_word not in self.unknown_words
+                    and self.private_dict_file is not None
```

Comment:
Isn't this a change in behaviour?

Reply:
If ``self.private_dict_file is None`` the check() call would crash pylint. It's never happening but you have to read the whole class workflow to realize that, and mypy can't.  I'll do a follow-up refactor to initialize thing in the constructor where it should be clearer.